Bug: Hidden (dot) directories aren't globbed in typescript-estree's projects globs

### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Issue Description

When `parserOptions.project` includes a `*` wildcard, the call to `globby` doesn't include `dot: true`.

https://github.com/typescript-eslint/typescript-eslint/blob/4ab9bd7f7c1812df4371d1fd3202969c1039e8a7/packages/typescript-estree/src/parseSettings/createParseSettings.ts#L188-L190

That means directories like `.test` won't have TSConfigs inside them found.

From this `projects` list:

```plaintext
project: [
  './examples/*/tsconfig.json',
  './examples/*/*/tsconfig.json',
  './packages/*/tsconfig.json',
  './tsconfig.eslint.json',
],
```

...and `tsconfig.json`s at:

```plaintext
./examples/.test/ssg/tsconfig.json
./packages/next/tsconfig.json
./packages/react-query/tsconfig.json
./packages/server/tsconfig.json
./packages/client/tsconfig.json
./tsconfig.json
./tsconfig.eslint.json
```

```plaintext
/Users/josh/repos/repros/examples/.test/ssg/src/utils/trpc.ts
  0:0  error  Parsing error: ESLint was configured to run on `<tsconfigRootDir>/examples/.test/ssg/src/utils/trpc.ts` using `parserOptions.project`: 
- <tsconfigRootDir>/../../../../users/josh/repos/repros/tsconfig.eslint.json
- <tsconfigRootDir>/../../../../users/josh/repos/repros/packages/client/tsconfig.json
- <tsconfigRootDir>/../../../../users/josh/repos/repros/packages/next/tsconfig.json
- <tsconfigRootDir>/../../../../users/josh/repos/repros/packages/react-query/tsconfig.json
- <tsconfigRootDir>/../../../../users/josh/repos/repros/packages/server/tsconfig.json
```

We can see this in the wild with https://github.com/trpc/trpc/pull/3390. Which is running out of memory in CI 🙃 but you can repro locally at https://github.com/trpc/trpc/pull/3390/commits/3de2e91c4b92d2863a1a299e15373a08f59ac486.

### Reproduction Repository Link

https://github.com/JoshuaKGoldberg/repros/tree/b9f06eed94237ebd5a2180988f3ca084b2b6ad5c

### Repro Steps

1) clone the repo
2) `pnpm install`
3) `pnpm lint`

Alternately, running `pnpm lint` from `examples/.test/ssg` runs faster and also shows the issues.

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `5.48.0` |
| `@typescript-eslint/parser`            | `5.48.0` |
| `@typescript-eslint/scope-manager`     | `5.48.0` |
| `@typescript-eslint/typescript-estree` | `5.48.0` |
| `@typescript-eslint/type-utils`        | `5.48.0` |
| `@typescript-eslint/utils`             | `5.48.0` |
| `TypeScript`                           | `4.8.3` |
| `ESLint`                               | `8.30.0` |
| `node`                                 | `18.7.0` |

--- a/packages/typescript-estree/src/parseSettings/resolveProjectList.ts
+++ b/packages/typescript-estree/src/parseSettings/resolveProjectList.ts
@@ -98,6 +98,7 @@ export function resolveProjectList(
           ? []
           : globSync([...globProjects, ...projectFolderIgnoreList], {
               cwd: options.tsconfigRootDir,
+              dot: true,
             }),
       )
       .map(project =>

